examples/frame_prediction/js/entities/entities.js

```diff
@@ -0,0 +1,76 @@
+/************************************************************************************/
+/*                                                                                  */
+/*     Base entity that predicts its position based on the last update              */
+/*                                                                                  */
+/************************************************************************************/
+game.FramePredictionEntity = me.Entity.extend({
+   
+    init: function(x, y, settings) {
+        this._super(me.Entity, "init", [x, y , settings]);
+    },
+    
+    draw: function(renderer) {
+        // Predict the next position to draw at based on velocity
+        var updateInterval = (me.timer.getTime() - me.timer.lastUpdate) / ( (1000/me.sys.updatesPerSecond) );
```

Comment:
I'm trying to work out in my head the best API. Certainly open to suggestions, but I feel the rectangle is going to be less useful than a time delta for animations in the draw method. That would prioritize the time delta as the second parameter, and an optional rectangle as the third.

Another point for passing the time delta to the draw methods is that all animation code (animation sheet, tile animations, ...) can be moved into the draw method, where it really belongs. Without this change, running updates at a frequency different from drawing would cause animation frames to drop (low frequency updates), or waste CPU (high frequency updates).